src/card/CardSelectors.ts

```diff
@@ -1,6 +1,6 @@
 export const getDashboardTitle = (state: any) => state.dashboard.title;
 
-export const getReportState = (state: any, index: any) => {
+export const getReportState = (state: any, id: any) => {
   const { pagenumber } = state.dashboard.settings;
-  return state.dashboard.pages[pagenumber].reports[index];
+  return state.dashboard.pages[pagenumber].reports.find((o) => o.id === id);
```

Comment:
i have a crazy idea here but you will hate me

Reply:
Not crazy.... soon to be implemented.